honeycombio/honeycomb-opentelemetry-web
# feat: Add Experimental User Interaction Instrumentation

supercedes https://github.com/honeycombio/honeycomb-opentelemetry-web/pull/243

## Which problem is this PR solving?
Upstream's `UserInteractionInstrumentation` does not play well in React apps with many nested event handlers. This adds an experimental alternative `UserInteractionInstrumentation`.

## Short description of the changes
Sets up an event listener at the top level instead of wrapping the global `addEventListener`. 

Smoke tests to come in a later PR. 

## How to verify that this has the expected result
Run the example react app, click on some stuff. 

The app looks like this 
![image](https://github.com/user-attachments/assets/cb02dcac-f423-4bf3-818e-4feb6a2338ac)

Clicks in the blue header should not create any spans.
Likewise, clicks below the outlined div should not create any spans.

Clicks anywhere in the middle div-- on the div itself, or on either button --should create spans.

You should be able to see the spans in the console log.
